core/src/View/Kokkos_ViewCtor.hpp

```diff
@@ -234,13 +233,14 @@ struct ViewCtorProp : public ViewCtorProp<void, P>... {
   // of confused. To work around this, we just use a couple of alias templates
   // that amount to the same thing.
   template <typename... Args>
-  ViewCtorProp(view_ctor_prop_args<Args...> const &arg)
+  ViewCtorProp([[maybe_unused]] view_ctor_prop_args<Args...> const &arg)
       : view_ctor_prop_base<Args>(
-            static_cast<view_ctor_prop_base<Args> const &>(arg))... {
-    // Suppress an unused argument warning that (at least at one point) would
-    // show up if sizeof...(Args) == 0
-    (void)arg;
-  }
+            static_cast<view_ctor_prop_base<Args> const &>(arg))... {}
+
+  template <typename... Args>
+  ViewCtorProp(view_ctor_prop_args<Args...> &&arg)
```

Comment:
Could these 2 constructors actually be merged into one ?

Reply:
Reverted.